supabase/migrations/20251223184411_encrypt-all-sensitive-data.sql

```diff
@@ -0,0 +1,1126 @@
+
+-- Add encrypted_data column to store all sensitive fields as encrypted JSON
+alter table wallet.cashu_receive_quotes add column encrypted_data text not null;
+alter table wallet.cashu_receive_quotes add column payment_hash text not null;
+
+-- Add quote_id_hash column to allow looking up quotes by their mint quote_id (hashed for privacy)
+alter table wallet.cashu_receive_quotes add column quote_id_hash text not null;
+
+-- Create unique index on quote_id_hash to make sure we don't have duplicate quotes with the same mint quote_id
+create unique index cashu_receive_quotes_quote_id_hash_key 
+  on wallet.cashu_receive_quotes using btree (quote_id_hash) 
+  where quote_id_hash is not null;
+
+-- Drop existing function signatures before recreating with new params
+drop function if exists wallet.create_cashu_receive_quote(uuid, uuid, numeric, text, text, text, text, timestamp with time zone, text, text, text, text, text, numeric);
+drop function if exists wallet.process_cashu_receive_quote_payment(uuid, text, integer[]);
+
+create or replace function wallet.create_cashu_receive_quote(
+  p_user_id uuid,
+  p_account_id uuid,
+  p_currency text,
+  p_unit text,
+  p_expires_at timestamp with time zone,
+  p_state text,
+  p_locking_derivation_path text,
+  p_receive_type text,
+  p_encrypted_transaction_details text,
+  p_encrypted_data text,
+  p_quote_id_hash text,
+  p_payment_hash text
+)
+returns wallet.cashu_receive_quotes
+language plpgsql
+as $function$
+declare
+  v_transaction_type text;
+  v_transaction_state text;
+  v_transaction_id uuid;
+  v_quote wallet.cashu_receive_quotes;
+begin
+  -- Map receive type to transaction type
+  v_transaction_type := case p_receive_type
+    when 'LIGHTNING' then 'CASHU_LIGHTNING'
+    when 'TOKEN' then 'CASHU_TOKEN'
+    else null
+  end;
+
+  if v_transaction_type is null then
+    raise exception 
+      using
+        hint = 'INVALID_ARGUMENT',
+        message = 'Unsupported receive type',
+        detail = format('Expected one of: LIGHTNING, TOKEN. Value provided: %s', p_receive_type);
+  end if;
+
+  -- We create token receives as pending because the lightning payment on the sender
+  -- side will be triggered by the receiver, so we know it should get paid.
+  -- For lightning, we create a draft transaction record because its not guaranteed that
+  -- the invoice will ever be paid.
+  v_transaction_state := case v_transaction_type
+    when 'CASHU_TOKEN' then 'PENDING'
+    else 'DRAFT'
+  end;
+
+  insert into wallet.transactions (
+    user_id,
+    account_id,
+    direction,
+    type,
+    state,
+    currency,
+    encrypted_transaction_details
+  ) values (
+    p_user_id,
+    p_account_id,
+    'RECEIVE',
+    v_transaction_type,
+    v_transaction_state,
+    p_currency,
+    p_encrypted_transaction_details
+  ) returning id into v_transaction_id;
+
+  insert into wallet.cashu_receive_quotes (
+    user_id,
+    account_id,
+    currency,
```

Comment:
should currency and unit also be encrypted, or is it kind of pointless since they can be seen from related account anyway?